### PR TITLE
⚡️📢☁️( Adds elements and logic to separate soundcloud embeds from embeds*)

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
Adds elements and logic to seperate soundcloud *embeds* from embeds*

*This logic will need repeating for audio elements.

## What is the value of this and can you measure success?

dcr amp 

### Does this affect other platforms?

- [X] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 1]
```diff
@@ -23,7 +25,8 @@ case class ImageSource(weighting: String, srcSet: Seq[SrcSet])
 case class AudioBlockElement(assets: Seq[AudioAsset]) extends PageElement
 case class GuVideoBlockElement(assets: Seq[VideoAsset], imageMedia: ImageMedia, data: Map[String, String], role: Role) extends PageElement
 case class VideoBlockElement(data: Map[String, String], role: Role) extends PageElement
-case class EmbedBlockElement(html: Option[String], safe: Option[Boolean], alt: Option[String], isMandatory: Option[Boolean]) extends PageElement
+case class EmbedBlockElement(html: String, safe: Option[Boolean], alt: Option[String], isMandatory: Boolean) extends PageElement
+case class SoundcloudBlockElement(html: String, track: Option[String], playlist: Option[String], isMandatory: Boolean) extends PageElement
```
Comment (AWare) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala]: Maybe we want `id` and some kind of track/playlist flag instead https://github.com/guardian/dotcom-rendering/pull/317#discussion_r240304479

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 2]
```diff
@@ -189,6 +192,43 @@ object PageElement {
...
+
+  private def extractEmbed(element: ApiBlockElement): Option[PageElement] = {
+      for {
+      d <- element.embedTypeData
+      html <- d.html
+      mandatory = d.isMandatory.getOrElse(false)
+      e = EmbedBlockElement(html, d.safeEmbedCode, d.alt, mandatory)
+      s = extractSoundcloud(html, mandatory)
+    } yield {
+      s getOrElse e
```
Comment (philmcmahon) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala]: can you use some more descriptive names maybe? dataType, embedBlock etc.

Comment (philmcmahon) [reply common/app/model/dotcomrendering/pageElements/PageElement.scala]: (also in extractAudio)

File: common/app/views/support/cleaner/AmpSoundcloud.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  def getSoundCloudElement(document: Document, iframeElement: Element): Option[Element] = {
+    val trackId = AmpSoundcloud.getTrackIdFromUrl(iframeElement.attr("src"))
+    val playlistId = AmpSoundcloud.getPlaylistIdFromUrl(iframeElement.attr("src"))
+
+    (trackId, playlistId) match {
+      case (Some(id),_) => Some(AmpSoundcloud.createElementFromTrack(document, id))
+      case (_,Some(id)) => Some(AmpSoundcloud.createElementFromPlaylist(document, id))
+      case _ => None
```
Comment (philmcmahon) [inline common/app/views/support/cleaner/AmpSoundcloud.scala]: nice 💅

**Review by philmcmahon — APPROVED** [review]

**Pull request merged**